src/cfnlint/data/AdditionalSpecs/Exclusive.json

```diff
@@ -59,6 +59,12 @@
         "MasterUserPassword",
         "StorageEncrypted"
       ],
+      "DBSnapshotIdentifier": [
```

Comment:
This used to be in this spec but there was a trick with it.  All the properties can be specified together as long as the DBSnapshotIdentifier is an empty string.

`If the property contains a value (which is not an empty string), AWS CloudFormation creates a database based on the specified snapshot. If no snapshot with the given name exists, AWS CloudFormation can not create the database and will roll back the stack.`

#229 is for creating a separate rule for this
#227 was the original issue

It haven't done a good job testing some of this as I think @farski tried to point out.  Can you specify exclusive attributes together if the value is an empty string or AWS::NoValue?  It could be that the RDS team was the only one to point this out in the documentation or it could be that they are the only ones that support this scenario.  Either way it would be worth a little more digging on our part.